src/Domain/Creators/Sphere.hpp

```diff
@@ -38,15 +39,82 @@ class CoordinateMap;
 }  // namespace domain
 /// \endcond
 
-namespace domain {
-namespace creators {
-/// Create a 3D Domain in the shape of a sphere consisting of six wedges
-/// and a central cube. For an image showing how the wedges are aligned in
-/// this Domain, see the documentation for Shell.
-///
-/// The inner cube is a BulgedCube except if the inner cube sphericity is
-/// exactly 0. Then an Equiangular or Affine map is used (depending on if it's
-/// equiangular or not) to avoid a root find in the BulgedCube map.
+namespace domain::creators {
+
+namespace detail {
+
+/// Options for excising the interior of the sphere
+struct Excision {
+  static constexpr Options::String help = {
+      "Excise the interior of the sphere, leaving a spherical shell."};
+  template <typename BoundaryConditionsBase>
+  struct BoundaryCondition {
+    static std::string name() { return "ExciseWithBoundaryCondition"; }
+    using type = std::unique_ptr<BoundaryConditionsBase>;
+    static constexpr Options::String help = {
+        "The boundary condition to impose on the excision surface."};
+  };
+  struct NoBoundaryCondition {
+    static std::string name() { return "Excise"; }
+    using type = bool;
+    static constexpr Options::String help = {
+        "Specify 'True' to excise the interior of the sphere. Instead of "
+        "specifying 'False', replace with the 'FillWithSphericity' option."};
+  };
```

Comment:
Is this an option with only one valid value?  Why do you need it?

Reply:
If boundary conditions are disabled (e.g. in ExportCoordinates), then there are two possibilities: (1) Fill interior with cube of specified sphericity, (2) just excise (no further options). It looks like this in the input file:

```yaml
Interior:
  FillWithSphericity: 0.
# or
Interior:
  Excise: True
```

This is the cleanest approach I've found for the "edge case" when boundary conditions are disabled. For the case when boundary conditions are enabled it looks like this:

```yaml
Interior:
  FillWithSphericity: 0.
# or
Interior:
  ExciseWithBoundaryCondition: # ...
```

Reply:
Maybe use `Auto` with "Excise" as the empty option, or add a custom parse function so the excision class can be created from just the string "Excise"?

Reply:
I won't approve it as-is, but I'll step back and allow one of the other core-devs to do so if they are happy with it.

Reply:
I looked at this again and can't find a nice alternative to the current solution. Parsing as `Auto` gets quite complicated as well because it has to go though another constructor etc, and I think adding all that code isn't worth it for this edge case with no boundary conditions.

Reply:
My concern with all this is that after spending 5 minutes reading through all the help strings I still am not completely sure I know what's going on or what the input file will look like. Is there a way to have the `variant`-based selection give a name:
```
Interior:
  Cube:
    Sphericity: 0.8
```
vs
```
Interior:
  Excise:
    BoundaryCondition:
```
vs
```
Interior:
  Excise:
```

While I agree it's an edge case, if I'm confused, I'm worried others will be too. Since we special case things on whether or not there are options, isn't making this a `basic_option` hindering us from resolving this more cleanly? As in, can we actually template `Interior` on a bool of whether or not we need the boundary condition? Eg
```cpp
template <bool Bc>
  struct Interior {
    struct ExciseInnerCube {};
    using type = tmpl::conditional_t<Bc, std::variant<Excision, InnerCube>, Options::Auto<double, ExciseInnerCube>>;
    static std::string name() { return Bc ? "Interior" : "InnerCubeSphericity"; }
    static constexpr Options::String help = {. // Need swap docs?
        "Sphericity of inner cube. A sphericity of 0 uses a product of 1D maps "
        "as the map in the center. A sphericity > 0 uses a BulgedCube. A "
        "sphericity of exactly 1 is not allowed. See BulgedCube docs for why."};
  };
```
If the documentation is annoying to swap out, we could just specialize on the two cases.

Hopefully this gives a reasonable solution to @wthrowe's concerns that's also not very tedious to implement.